compiler error from source in linux cluster

I want to run your tts model, I compiler sherpa-onnx follow the steps:
```git clone https://github.com/k2-fsa/sherpa-onnx
cd sherpa-onnx
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release ..
make -j6
```
however, when I run this command: ```cmake -DCMAKE_BUILD_TYPE=Release ..``` ,it occurs the below errors:
```
-- The C compiler identification is GNU 8.5.0
-- The CXX compiler identification is GNU 8.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_INSTALL_PREFIX: /usr/local
-- BUILD_SHARED_LIBS OFF
-- SHERPA_ONNX_ENABLE_PYTHON OFF
-- SHERPA_ONNX_ENABLE_TESTS OFF
-- SHERPA_ONNX_ENABLE_CHECK OFF
-- SHERPA_ONNX_ENABLE_PORTAUDIO ON
-- SHERPA_ONNX_ENABLE_JNI OFF
-- SHERPA_ONNX_ENABLE_C_API ON
-- SHERPA_ONNX_ENABLE_WEBSOCKET ON
-- SHERPA_ONNX_ENABLE_GPU OFF
-- C++ Standard version: 14
-- Looking for C++ include alsa/asoundlib.h
-- Looking for C++ include alsa/asoundlib.h - not found
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Looking for C++ include execinfo.h
-- Looking for C++ include execinfo.h - found
-- Link libstdc++ statically
-- Downloading kaldi-native-fbank from https://github.com/csukuangfj/kaldi-native-fbank/archive/refs/tags/v1.18.5.tar.gz
-- kaldi-native-fbank is downloaded to /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldi_native_fbank-src
-- kaldi-native-fbank's binary dir is /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldi_native_fbank-build
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_EXPORT_COMPILE_COMMANDS:
-- BUILD_SHARED_LIBS: OFF
-- KALDI_NATIVE_FBANK_BUILD_TESTS: OFF
-- KALDI_NATIVE_FBANK_BUILD_PYTHON: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- CMAKE_CXX_FLAGS:   -static-libstdc++ -static-libgcc
-- CMAKE_INSTALL_PREFIX: /usr/local
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - not found
-- Looking for C++ include execinfo.h
-- Looking for C++ include execinfo.h - not found
-- Disable building Python
-- Downloading kaldi-decoder from https://github.com/k2-fsa/kaldi-decoder/archive/refs/tags/v0.2.3.tar.gz
-- kaldi-decoder is downloaded to /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldi_decoder-src
-- kaldi-decoder's binary dir is /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldi_decoder-build
-- CMAKE_BUILD_TYPE: Release
-- Downloading kaldifst from https://github.com/k2-fsa/kaldifst/archive/refs/tags/v1.7.9.tar.gz
-- kaldifst is downloaded to /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldifst-src
-- kaldifst's binary dir is /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/kaldifst-build
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_INSTALL_PREFIX: /usr/local
-- BUILD_SHARED_LIBS OFF
-- C++ Standard version: 14
-- CMAKE_CXX_FLAGS:   -static-libstdc++ -static-libgcc
-- Downloading openfst from https://github.com/kkm000/openfst/archive/refs/tags/win/1.6.5.1.tar.gz
-- openfst is downloaded to /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/openfst-src
-- Found the following ICU libraries:
--   data (required)
--   i18n (required)
--   io (required)
--   test (required)
--   tu (required)
--   uc (required)
-- Found ICU: /usr/include (found version "60.3")
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11")
-- Downloading eigen from https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
-- eigen is downloaded to /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/eigen-src
-- eigen's binary dir is /mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/_deps/eigen-build
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Failed
-- Performing Test COMPILER_SUPPORT_std=cpp03
-- Performing Test COMPILER_SUPPORT_std=cpp03 - Failed
-- Performing Test standard_math_library_linked_to_automatically
-- Performing Test standard_math_library_linked_to_automatically - Failed
-- Performing Test standard_math_library_linked_to_as_m
-- Performing Test standard_math_library_linked_to_as_m - Failed
CMake Error at build/_deps/eigen-src/CMakeLists.txt:108 (message):
  Can't link to the standard math library.  Please report to the Eigen
  developers, telling them about your platform.


-- Configuring incomplete, errors occurred!
See also "/mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/CMakeFiles/CMakeOutput.log".
See also "/mntnfs/lee_data1/maduo/eggs_exp_audiodata/sherpa-onnx/build/CMakeFiles/CMakeError.log".
```

cmake version is as follows:
```
cmake --version
cmake version 3.18.6

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

I meet the same problem
My OS is ubuntu18.04

-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
I tried switching the GNU version to 12.1.0, but it still didn't work
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0

```
[ 92%] Linking CXX executable ../../bin/sherpa-onnx-offline-parallel
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `OrtApis::ReleasePrepackedWeightsContainer(OrtPrepackedWeightsContainer*)':
onnxruntime_c_api.cc:(.text._ZN7OrtApis32ReleasePrepackedWeightsContainerEP28OrtPrepackedWeightsContainer+0x253): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `std::unique_ptr<OrtValue, std::default_delete<OrtValue> >& absl::lts_20240116::inlined_vector_internal::Storage<std::unique_ptr<OrtValue, std::default_delete<OrtValue> >, 6ul, std::allocator<std::unique_ptr<OrtValue, std::default_delete<OrtValue> > > >::EmplaceBackSlow<std::unique_ptr<OrtValue, std::default_delete<OrtValue> > >(std::unique_ptr<OrtValue, std::default_delete<OrtValue> >&&) [clone .isra.0]':
onnxruntime_c_api.cc:(.text._ZN4absl12lts_2024011623inlined_vector_internal7StorageISt10unique_ptrI8OrtValueSt14default_deleteIS4_EELm6ESaIS7_EE15EmplaceBackSlowIJS7_EEERS7_DpOT_.isra.0+0xc8): undefined reference to `__libc_single_threaded'
onnxruntime_c_api.cc:(.text._ZN4absl12lts_2024011623inlined_vector_internal7StorageISt10unique_ptrI8OrtValueSt14default_deleteIS4_EELm6ESaIS7_EE15EmplaceBackSlowIJS7_EEERS7_DpOT_.isra.0+0x13c): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `std::default_delete<OrtValue>::operator()(OrtValue*) const [clone .constprop.0]':
onnxruntime_c_api.cc:(.text._ZNKSt14default_deleteI8OrtValueEclEPS0_.constprop.0+0x18): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `OrtApis::ReleaseValue(OrtValue*)':
onnxruntime_c_api.cc:(.text._ZN7OrtApis12ReleaseValueEP8OrtValue+0x1c): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o):onnxruntime_c_api.cc:(.text._ZN7OrtApis16ReleaseIoBindingEP12OrtIoBinding+0x75): more undefined references to `__libc_single_threaded' follow
../../_deps/onnxruntime-src/lib/libonnxruntime.a(implementation.cc.o): In function `std::_Rb_tree<int, std::pair<int const, std::shared_ptr<onnx::FunctionProto> >, std::_Select1st<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >, std::less<int>, std::allocator<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > > >::_M_erase(std::_Rb_tree_node<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >*) [clone .isra.0]':
implementation.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::SinceVersion(int)':
schema.cc:(.text._ZN4onnx8OpSchema12SinceVersionEi+0x37a): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(char const*, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyEPKci+0x161): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EEi+0x113): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, std::vector<onnx::OperatorSetIdProto, std::allocator<onnx::OperatorSetIdProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EERKS1_INS_18OperatorSetIdProtoESaIS7_EEi+0x193): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(checker.cc.o):checker.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): more undefined references to `__libc_single_threaded' follow
../../_deps/onnxruntime-src/lib/libonnxruntime.a(implementation.cc.o): In function `std::_Rb_tree<int, std::pair<int const, std::shared_ptr<onnx::FunctionProto> >, std::_Select1st<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >, std::less<int>, std::allocator<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > > >::_M_erase(std::_Rb_tree_node<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >*) [clone .isra.0]':
implementation.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::SinceVersion(int)':
schema.cc:(.text._ZN4onnx8OpSchema12SinceVersionEi+0x37a): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(char const*, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyEPKci+0x161): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EEi+0x113): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, std::vector<onnx::OperatorSetIdProto, std::allocator<onnx::OperatorSetIdProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EERKS1_INS_18OperatorSetIdProtoESaIS7_EEi+0x193): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(checker.cc.o):checker.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): more undefined references to `__libc_single_threaded' follow
../../_deps/onnxruntime-src/lib/libonnxruntime.a(implementation.cc.o): In function `std::_Rb_tree<int, std::pair<int const, std::shared_ptr<onnx::FunctionProto> >, std::_Select1st<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >, std::less<int>, std::allocator<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > > >::_M_erase(std::_Rb_tree_node<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >*) [clone .isra.0]':
implementation.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::SinceVersion(int)':
schema.cc:(.text._ZN4onnx8OpSchema12SinceVersionEi+0x37a): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(char const*, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyEPKci+0x161): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EEi+0x113): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, std::vector<onnx::OperatorSetIdProto, std::allocator<onnx::OperatorSetIdProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EERKS1_INS_18OperatorSetIdProtoESaIS7_EEi+0x193): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(checker.cc.o):checker.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): more undefined references to `__libc_single_threaded' follow
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `OrtApis::ReleasePrepackedWeightsContainer(OrtPrepackedWeightsContainer*)':
onnxruntime_c_api.cc:(.text._ZN7OrtApis32ReleasePrepackedWeightsContainerEP28OrtPrepackedWeightsContainer+0x253): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `std::unique_ptr<OrtValue, std::default_delete<OrtValue> >& absl::lts_20240116::inlined_vector_internal::Storage<std::unique_ptr<OrtValue, std::default_delete<OrtValue> >, 6ul, std::allocator<std::unique_ptr<OrtValue, std::default_delete<OrtValue> > > >::EmplaceBackSlow<std::unique_ptr<OrtValue, std::default_delete<OrtValue> > >(std::unique_ptr<OrtValue, std::default_delete<OrtValue> >&&) [clone .isra.0]':
onnxruntime_c_api.cc:(.text._ZN4absl12lts_2024011623inlined_vector_internal7StorageISt10unique_ptrI8OrtValueSt14default_deleteIS4_EELm6ESaIS7_EE15EmplaceBackSlowIJS7_EEERS7_DpOT_.isra.0+0xc8): undefined reference to `__libc_single_threaded'
onnxruntime_c_api.cc:(.text._ZN4absl12lts_2024011623inlined_vector_internal7StorageISt10unique_ptrI8OrtValueSt14default_deleteIS4_EELm6ESaIS7_EE15EmplaceBackSlowIJS7_EEERS7_DpOT_.isra.0+0x13c): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `std::default_delete<OrtValue>::operator()(OrtValue*) const [clone .constprop.0]':
onnxruntime_c_api.cc:(.text._ZNKSt14default_deleteI8OrtValueEclEPS0_.constprop.0+0x18): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o): In function `OrtApis::ReleaseValue(OrtValue*)':
onnxruntime_c_api.cc:(.text._ZN7OrtApis12ReleaseValueEP8OrtValue+0x1c): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(onnxruntime_c_api.cc.o):onnxruntime_c_api.cc:(.text._ZN7OrtApis16ReleaseIoBindingEP12OrtIoBinding+0x75): more undefined references to `__libc_single_threaded' follow
collect2: error: ld returned 1 exit status
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts.dir/build.make:92: recipe for target 'bin/sherpa-onnx-offline-tts' failed
make[2]: *** [bin/sherpa-onnx-offline-tts] Error 1
CMakeFiles/Makefile2:2349: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
collect2: error: ld returned 1 exit status
../../_deps/onnxruntime-src/lib/libonnxruntime.a(implementation.cc.o): In function `std::_Rb_tree<int, std::pair<int const, std::shared_ptr<onnx::FunctionProto> >, std::_Select1st<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >, std::less<int>, std::allocator<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > > >::_M_erase(std::_Rb_tree_node<std::pair<int const, std::shared_ptr<onnx::FunctionProto> > >*) [clone .isra.0]':
implementation.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): undefined reference to `__libc_single_threaded'
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-vad-microphone.dir/build.make:108: recipe for target 'bin/sherpa-onnx-vad-microphone' failed
make[2]: *** [bin/sherpa-onnx-vad-microphone] Error 1
CMakeFiles/Makefile2:2245: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-vad-microphone.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-vad-microphone.dir/all] Error 2
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::SinceVersion(int)':
schema.cc:(.text._ZN4onnx8OpSchema12SinceVersionEi+0x37a): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(char const*, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyEPKci+0x161): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EEi+0x113): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(schema.cc.o): In function `onnx::OpSchema::FunctionBody(std::vector<onnx::NodeProto, std::allocator<onnx::NodeProto> > const&, std::vector<onnx::OperatorSetIdProto, std::allocator<onnx::OperatorSetIdProto> > const&, int)':
schema.cc:(.text._ZN4onnx8OpSchema12FunctionBodyERKSt6vectorINS_9NodeProtoESaIS2_EERKS1_INS_18OperatorSetIdProtoESaIS7_EEi+0x193): undefined reference to `__libc_single_threaded'
../../_deps/onnxruntime-src/lib/libonnxruntime.a(checker.cc.o):checker.cc:(.text._ZNSt8_Rb_treeIiSt4pairIKiSt10shared_ptrIN4onnx13FunctionProtoEEESt10_Select1stIS6_ESt4lessIiESaIS6_EE8_M_eraseEPSt13_Rb_tree_nodeIS6_E.isra.0+0x33): more undefined references to `__libc_single_threaded' follow
collect2: error: ld returned 1 exit status
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/build.make:108: recipe for target 'bin/sherpa-onnx-microphone-offline' failed
make[2]: *** [bin/sherpa-onnx-microphone-offline] Error 1
CMakeFiles/Makefile2:2280: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/all] Error 2
collect2: error: ld returned 1 exit status
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts-play.dir/build.make:108: recipe for target 'bin/sherpa-onnx-offline-tts-play' failed
make[2]: *** [bin/sherpa-onnx-offline-tts-play] Error 1
CMakeFiles/Makefile2:2451: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts-play.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-tts-play.dir/all] Error 2
collect2: error: ld returned 1 exit status
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/build.make:108: recipe for target 'bin/sherpa-onnx-microphone' failed
make[2]: *** [bin/sherpa-onnx-microphone] Error 1
CMakeFiles/Makefile2:2315: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/all] Error 2
collect2: error: ld returned 1 exit status
sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-parallel.dir/build.make:92: recipe for target 'bin/sherpa-onnx-offline-parallel' failed
make[2]: *** [bin/sherpa-onnx-offline-parallel] Error 1
CMakeFiles/Makefile2:2383: recipe for target 'sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-parallel.dir/all' failed
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline-parallel.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

When I use GNU 12.1.0
```
cmake -DBUILD_SHARED_LIBS=ON ..
make -j4
```
still not work

When  i use GNU 7.5.0 , it's work

--- a/.github/workflows/linux.yaml
+++ b/.github/workflows/linux.yaml
@@ -57,13 +57,13 @@ jobs:
         # 7-11 for ubuntu 20.04
         # 9-12 for ubuntu 22.04
         gcc_version: ["7", "8", "9", "10", "11", "12", "13"]
-        # exclude:
-        #   - os: ubuntu-20.04
-        #     gcc_version: "12"
-        #   - os: ubuntu-22.04
-        #     gcc_version: "7"
-        #   - os: ubuntu-22.04
-        #     gcc_version: "8"
+        exclude:
+          - os: ubuntu-20.04
+            gcc_version: "12"
+          - os: ubuntu-22.04
+            gcc_version: "7"
+          - os: ubuntu-22.04
+            gcc_version: "8"
 
     steps:
       - uses: actions/checkout@v4

--- a/.github/workflows/linux.yaml
+++ b/.github/workflows/linux.yaml
@@ -45,24 +45,50 @@ concurrency:
 
 jobs:
   linux:
-    name: ${{ matrix.build_type }} ${{ matrix.shared_lib }}
+    name: ${{ matrix.os }} ${{ matrix.build_type }} ${{ matrix.shared_lib }} ${{ matrix.gcc_version }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-20.04, ubuntu-22.04]
         build_type: [Release, Debug]
         shared_lib: [ON, OFF]
+        # see https://github.com/egor-tensin/setup-gcc
+        # 7-11 for ubuntu 20.04
+        # 9-12 for ubuntu 22.04
+        gcc_version: ["7", "8", "9", "10", "11", "12", "13"]
+        # exclude:
+        #   - os: ubuntu-20.04
+        #     gcc_version: "12"
+        #   - os: ubuntu-22.04
+        #     gcc_version: "7"
+        #   - os: ubuntu-22.04
+        #     gcc_version: "8"
 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Set up GCC ${{ matrix.gcc_version }}
+        uses: egor-tensin/setup-gcc@v1
+        with:
+          version: ${{ matrix.gcc_version }}
+          platform: x64
+
+      - name: Display gcc version ${{ matrix.gcc_version }}
+        shell: bash
+        run: |
+          which gcc
+          gcc --version
+
+          which g++
+          g++ --version
+
       - name: ccache
         uses: hendrikmuhs/ccache-action@v1.2
         with:
-          key: ${{ matrix.os }}-${{ matrix.build_type }}-shared-${{ matrix.shared_lib }}
+          key: ${{ matrix.os }}-${{ matrix.build_type }}-shared-${{ matrix.shared_lib }}-gcc-${{ matrix.gcc_version }}
 
       - name: Configure CMake
         shell: bash

--- a/cmake/onnxruntime-linux-x86_64-static.cmake
+++ b/cmake/onnxruntime-linux-x86_64-static.cmake
@@ -14,19 +14,19 @@ if(BUILD_SHARED_LIBS)
   message(FATAL_ERROR "This file is for building static libraries. BUILD_SHARED_LIBS: ${BUILD_SHARED_LIBS}")
 endif()
 
-set(onnxruntime_URL  "https://github.com/csukuangfj/onnxruntime-libs/releases/download/v1.17.0/onnxruntime-linux-x64-static_lib-1.17.0.zip")
-set(onnxruntime_URL2 "https://huggingface.co/csukuangfj/onnxruntime-libs/resolve/main/onnxruntime-linux-x64-static_lib-1.17.0.zip")
-set(onnxruntime_HASH "SHA256=137a5e5a9195c74452f3b772533545441ef0933a154fa006f2a13da8f956907d")
+set(onnxruntime_URL  "https://github.com/csukuangfj/onnxruntime-libs/releases/download/v1.17.0/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip")
+set(onnxruntime_URL2 "https://huggingface.co/csukuangfj/onnxruntime-libs/resolve/main/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip")
+set(onnxruntime_HASH "SHA256=4217302eac0d645b4c2b5fa13d1627c0384d2531615b4a51a52326c01568cc0c")
 
 # If you don't have access to the Internet,
 # please download onnxruntime to one of the following locations.
 # You can add more if you want.
 set(possible_file_locations
-  $ENV{HOME}/Downloads/onnxruntime-linux-x64-static_lib-1.17.0.zip
-  ${CMAKE_SOURCE_DIR}/onnxruntime-linux-x64-static_lib-1.17.0.zip
-  ${CMAKE_BINARY_DIR}/onnxruntime-linux-x64-static_lib-1.17.0.zip
-  /tmp/onnxruntime-linux-x64-static_lib-1.17.0.zip
-  /star-fj/fangjun/download/github/onnxruntime-linux-x64-static_lib-1.17.0.zip
+  $ENV{HOME}/Downloads/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip
+  ${CMAKE_SOURCE_DIR}/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip
+  ${CMAKE_BINARY_DIR}/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip
+  /tmp/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip
+  /star-fj/fangjun/download/github/onnxruntime-linux-x64-static_lib-1.17.0-gcc-8.zip
 )
 
 foreach(f IN LISTS possible_file_locations)

--- a/cmake/piper-phonemize.cmake
+++ b/cmake/piper-phonemize.cmake
@@ -1,18 +1,18 @@
 function(download_piper_phonemize)
   include(FetchContent)
 
-  set(piper_phonemize_URL  "https://github.com/csukuangfj/piper-phonemize/archive/6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip")
+  set(piper_phonemize_URL  "https://github.com/csukuangfj/piper-phonemize/archive/dc6b5f4441bffe521047086930b0fc12686acd56.zip")
   set(piper_phonemize_URL2 "")
-  set(piper_phonemize_HASH "SHA256=6fbacf540b03f00d1386bb372fb7090e3bb852bd019d74e615d3f161f728bc93")
+  set(piper_phonemize_HASH "SHA256=b9faa04204b1756fa455a962abb1f037041c040133d55be58d11f11ab9b3ce14")
 
   # If you don't have access to the Internet,
   # please pre-download kaldi-decoder
   set(possible_file_locations
-    $ENV{HOME}/Downloads/piper-phonemize-6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip
-    ${CMAKE_SOURCE_DIR}/piper-phonemize-6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip
-    ${CMAKE_BINARY_DIR}/piper-phonemize-6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip
-    /tmp/piper-phonemize-6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip
-    /star-fj/fangjun/download/github/piper-phonemize-6383e46b62e94c5cafc0c6a6212249ed8b9ed8d0.zip
+    $ENV{HOME}/Downloads/piper-phonemize-dc6b5f4441bffe521047086930b0fc12686acd56.zip
+    ${CMAKE_SOURCE_DIR}/piper-phonemize-dc6b5f4441bffe521047086930b0fc12686acd56.zip
+    ${CMAKE_BINARY_DIR}/piper-phonemize-dc6b5f4441bffe521047086930b0fc12686acd56.zip
+    /tmp/piper-phonemize-dc6b5f4441bffe521047086930b0fc12686acd56.zip
+    /star-fj/fangjun/download/github/piper-phonemize-dc6b5f4441bffe521047086930b0fc12686acd56.zip
   )
 
   foreach(f IN LISTS possible_file_locations)

